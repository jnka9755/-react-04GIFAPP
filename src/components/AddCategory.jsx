import { useState } from 'react';
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const inputChange = ({target}) =>{
        setInputValue(target.value);
    }

    const addCategory = (event) =>{

        event.preventDefault();
        const value = inputValue.trim();

        if( value.length < 1 ) return;
        
        onNewCategory(value);
        setInputValue('');
    }

    return (
        <form onSubmit={ (event) => addCategory(event) } aria-label='form'>
            <input 
            type="text"
            placeholder="Buscar gifs"
            value={ inputValue }
            onChange={ (event) => inputChange(event) }
            />
        </form>
    )
}

AddCategory.propTypes ={
    onNewCategory: PropTypes.func.isRequired
}