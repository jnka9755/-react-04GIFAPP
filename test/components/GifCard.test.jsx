import { render, screen } from "@testing-library/react";
import { GifCard } from "../../src/components/GifCard";

describe('GifCard Test', () => { 

    const title = 'Test title';
    const url = 'http://localhost/Test';

    test('Test snapshot', () => { 

        const { container } = render (
            <GifCard
                title={ title }
                url={ url }
            />
        );
        
        expect( container ).toMatchSnapshot();

     });

     test('Test show image', () => { 

        render (
            <GifCard 
                title={ title }
                url={ url }
            />
        );

        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        const { src , alt } = screen.getByRole('img');

        expect( src ).toBe( url );
        expect( alt ).toBe( alt ); 
      });

      test('Test title', () => {

        render (
            <GifCard 
                title={ title }
                url={ url }
            />
        );

        expect( screen.getByText( title )).toBeTruthy();
      });
 });