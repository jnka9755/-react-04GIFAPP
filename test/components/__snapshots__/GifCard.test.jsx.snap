// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifCard Test Test snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test title"
      src="http://localhost/Test"
    />
    <p>
      Test title
    </p>
  </div>
</div>
`;
